tools/build_defs/repo/http.bzl

```diff
@@ -55,25 +55,29 @@ def _http_archive_impl(ctx):
     patch(ctx)
     workspace_and_buildfile(ctx)
 
+_HTTP_FILE_FORBIDDEN_NAMES = ["WORKSPACE", "BUILD", "BUILD.bazel"]
+
 _HTTP_FILE_BUILD = """
 package(default_visibility = ["//visibility:public"])
-
 filegroup(
     name = "file",
-    srcs = ["downloaded"],
+    srcs = ["%s"],
 )
 """
 
 def _http_file_impl(ctx):
     """Implementation of the http_file rule."""
+    if ctx.attr.downloaded_file_name in _HTTP_FILE_FORBIDDEN_NAMES:
+      ctx.fail("%s cannot be used as downloaded_file_name in http_file" % ctx.attr.downloaded_file_name)
+    downloaded_file_name = ctx.attr.downloaded_file_name or "downloaded"
     ctx.download(
         ctx.attr.urls,
-        "file/downloaded",
+        "file/" + downloaded_file_name,
         ctx.attr.sha256,
         ctx.attr.executable,
     )
     ctx.file("WORKSPACE", "workspace(name = \"{name}\")".format(name = ctx.name))
-    ctx.file("file/BUILD", _HTTP_FILE_BUILD)
+    ctx.file("file/BUILD", _HTTP_FILE_BUILD % downloaded_file_name)
```

Comment:
prefer `format` over positional interpolation for long strings.

Reply:
done